File: Current/Product/Production/Common/Common.Tasks/CaseContainerCollection.vb
Option Explicit On
Option Strict On

Public Class CaseContainerCollection
    Inherits CollectionBase

    Public Sub New()

    End Sub

    Public Sub New(ByVal value As CaseContainerCollection)
        Me.InnerList.AddRange(value)
    End Sub

    Public Sub New(ByVal value As CaseElement())
        Me.InnerList.AddRange(value)
    End Sub

    Public Function Add(ByVal item As CaseElement) As Integer
        Return MyBase.List.Add(item)
    End Function

    Public Sub AddRange(ByVal items As CaseElement())
        Me.InnerList.AddRange(items)
    End Sub

    Public Sub AddRange(ByVal items As CaseContainerCollection)
        Me.InnerList.AddRange(items)
    End Sub
    Public Function Contains(ByVal item As CaseElement) As Boolean
        Return Me.InnerList.Contains(item)
    End Function
    Public Sub CopyTo(ByVal array As CaseElement(), ByVal index As Integer)
        Me.InnerList.CopyTo(array, index)
    End Sub
    Public Function IndexOf(ByVal item As CaseElement) As Integer
        Return Me.InnerList.IndexOf(item)
    End Function
    Public Sub Insert(ByVal index As Integer, ByVal item As CaseElement)
        Me.InnerList.Insert(index, item)
    End Sub
    Public Sub Remove(ByVal item As CaseElement)
        Me.InnerList.Remove(item)
    End Sub

    Public Property Item(ByVal index As Integer) As CaseElement
        Get
            Return DirectCast(Me.InnerList(index), CaseElement)
        End Get
        Set(ByVal value As CaseElement)
            Me.InnerList(index) = value
        End Set
    End Property


End Class

File: tags/0.9.0.140/Product/Production/Nant/CIFactory.NAnt.Tasks/Types/CaseContainerCollection.cs
using System;
using System.Collections;

namespace CIFactory.NAnt.Types
{
    public class CaseContainerCollection : CollectionBase
    {
        #region Constructors

        public CaseContainerCollection(CaseContainerCollection value)
        {
            this.InnerList.AddRange(value);
        }

        public CaseContainerCollection(CaseElement[] value)
        {
            this.InnerList.AddRange(value);
        }

        public CaseContainerCollection()
        {
        }

        #endregion

        #region Properties

        public CaseElement this[int index]
        {
            get { return (CaseElement)this.InnerList[index]; }
            set { this.InnerList[index] = value; }
        }

        #endregion

        #region Public Methods

        public int Add(CaseElement item)
        {
            return base.List.Add(item);
        }

        public void AddRange(CaseContainerCollection items)
        {
            this.InnerList.AddRange(items);
        }

        public void AddRange(CaseElement[] items)
        {
            this.InnerList.AddRange(items);
        }

        public bool Contains(CaseElement item)
        {
            return this.InnerList.Contains(item);
        }

        public void CopyTo(CaseElement[] array, int index)
        {
            this.InnerList.CopyTo(array, index);
        }

        public int IndexOf(CaseElement item)
        {
            return this.InnerList.IndexOf(item);
        }

        public void Insert(int index, CaseElement item)
        {
            this.InnerList.Insert(index, item);
        }

        public void Remove(CaseElement item)
        {
            this.InnerList.Remove(item);
        }

        #endregion

    }
}